src/main/java/org/dependencytrack/notification/publisher/AbstractWebhookPublisher.java

```diff
@@ -55,8 +55,8 @@ public void publish(final String publisherName, final PebbleTemplate template, f
         try {
             credentials = getBasicAuthCredentials();
         } catch (PublisherException e) {
-            logger.warn("An error occurred during the retrieval of credentials needed for notification publication. Skipping notification");
-            return;
+            logger.warn("An error occurred during the retrieval of credentials needed for notification publication. Skipping notification", e);
+            throw e;
```

Comment:
What's the reason for this behavior change (re-throwing the exception)?

Reply:
I feel that the exception stacktrace should be part of the log. Multiple exceptions can arise from `getBasicAuthCredentials()`, including exceptions resulting from the retrieval and decryption of the stored jira password, and it would help the user to pinpoint the issue.

As for rethrowing the exception or silently returning, I leave it up to you to decide.

This isn't strictly in relation to the PR title, but is on a related subject (jira publisher) and doesn't grant a separated PR imo.

Reply:
I'm with you on the logging the stacktrace part, but re-throwing the exception may have some undesirable side effects. Best case the exception is just logged twice because it will be handled again by the "unhandled exception handler", worst case other operations may be negatively affected.